### PR TITLE
jnp.linalg: improve API documentation

### PR DESCRIPTION
**Pull request by jakevdp**
Part of #21461.

### CODE REVIEW

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/linalg.py [hunk 1]
```diff
@@ -64,9 +63,58 @@ def _H(x: ArrayLike) -> Array:
...
 def cholesky(a: ArrayLike, *, upper: bool = False) -> Array:
+  """Compute the Cholesky decomposition of a matrix.
+
+  JAX implementation of :func:`numpy.linalg.cholesky`.
+
+  The Cholesky decomposition of a matrix `A` is:
+
+  .. math::
+
+     A = U^HU = LL^H
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py]: Done

File: jax/_src/numpy/linalg.py [hunk 2]
```diff
@@ -64,9 +63,58 @@ def _H(x: ArrayLike) -> Array:
...
+  where `U` is an upper-triangular matrix and `L` is a lower-triangular matrix.
+
+  Args:
+    a: input array, representing a (batched) positive-definite hermitian matrix.
+      Must have shape ``(..., N, N)``.
+    upper: if True, compute the upper Cholesky decomposition `L`. if False
+      (default), compute the lower Cholesky decomposition `U`.
+
+  Returns:
+    array of shape ``(..., N, N)`` representing the cholesky decomposition
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py]: Done

File: jax/_src/numpy/linalg.py [hunk 3]
```diff
@@ -64,9 +63,58 @@ def _H(x: ArrayLike) -> Array:
...
+    of the input.
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py]: Done

File: jax/_src/numpy/linalg.py [hunk 4]
```diff
@@ -221,9 +379,40 @@ def matrix_power(a: ArrayLike, n: int) -> Array:
...
+  """Compute the rank of a matrix.
+
+  JAX implementation of :func:`numpy.linalg.matrix_rank`.
+
+  The rank is calculated via the Singular Value Decomposition (SVD), and determined
+  by the number of singular values greater than the specified tolerance.
+
+  Args:
+    a: array of shape ``(..., M, N)`` whose rank is to be computed.
+    tol: optional array of shape ``(...)`` specifyin the tolerance. Singular values
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py]: Done

File: jax/_src/numpy/linalg.py [hunk 5]
```diff
@@ -64,9 +63,58 @@ def _H(x: ArrayLike) -> Array:
...
+      Must have shape ``(..., N, N)``.
+    upper: if True, compute the upper Cholesky decomposition `L`. if False
+      (default), compute the lower Cholesky decomposition `U`.
+
+  Returns:
+    array of shape ``(..., N, N)`` representing the cholesky decomposition
+    of the input.
+
+  See also:
+    - :func:`jax.scipy.linalg.cholesky`: SciPy-stype Cholesky API
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/numpy/linalg.py]: Done

**Review by hawkinsp — APPROVED** [review]

**Review by hawkinsp — COMMENTED** [review]
Note lint failure.

**Pull request merged**